pkg/tsdb/frame_util.go

```diff
@@ -35,3 +37,67 @@ func convertTSDBTimePoint(point TimePoint) (t *time.Time, f *float64) {
 	}
 	return
 }
+
+// FrameToSeries converts a frame that is a valid time series as per data.TimeSeriesSchema()
+// to a TimeSeriesSlice.
+func FrameToSeries(frame *data.Frame) (TimeSeriesSlice, error) {
+	tsSchema := frame.TimeSeriesSchema()
+	if tsSchema.Type == data.TimeSeriesTypeNot {
+		return nil, fmt.Errorf("input frame is not recognized as a time series")
+	}
+	// If Long, make wide
+	if tsSchema.Type == data.TimeSeriesTypeLong {
+		var err error
+		frame, err = data.LongToWide(frame)
+		if err != nil {
+			return nil, err // TODO: Kyle needs to figure out error wrapping and get with times.
+		}
+		tsSchema = frame.TimeSeriesSchema()
+	}
+	seriesCount := len(tsSchema.ValueIndices)
+	seriesSlice := make(TimeSeriesSlice, 0, seriesCount)
+	timeField := frame.Fields[tsSchema.TimeIndex]
+	timeNullFloatSlice := make([]null.Float, timeField.Len())
+	for i := 0; i < timeField.Len(); i++ {
+		tStamp, err := timeField.FloatAt(i)
+		if err != nil {
+			return nil, err
+		}
+		timeNullFloatSlice[i] = null.FloatFrom(tStamp)
+	}
+	for _, fieldIdx := range tsSchema.ValueIndices {
+		field := frame.Fields[fieldIdx]
+		ts := &TimeSeries{
+			Name:   field.Name,
+			Tags:   field.Labels.Copy(),
+			Points: make(TimeSeriesPoints, field.Len()),
+		}
+		for rowIdx := 0; rowIdx < field.Len(); rowIdx++ {
+			val, err := field.FloatAt(rowIdx)
+			if err != nil {
+				return nil, err // TODO: wrap...
+			}
+			ts.Points[rowIdx] = TimePoint{
+				null.FloatFrom(val),
+				timeNullFloatSlice[rowIdx],
+			}
+		}
+		seriesSlice = append(seriesSlice, ts)
+	}
+
+	return seriesSlice, nil
+
+}
+
+// FramesFromBytes returns a data.Frame slice from marshalled arrow dataframes.
+func FramesFromBytes(bFrames [][]byte) ([]*data.Frame, error) {
+	frames := make([]*data.Frame, len(bFrames))
+	for i, bFrame := range bFrames {
+		var err error
+		frames[i], err = data.UnmarshalArrow(bFrame)
```

Comment:
any reason for not writing?
```go
frames[i], err := data.UnmarshalArrow(bFrame)
```

Reply:
reason is that it isn't valid go code :-)

Reply:
I was suspecting that. I should have tried it myself. :)